src/features/defi/providers/univ2/hooks/useUniV2LiquidityPool.ts

```diff
@@ -67,57 +82,142 @@ export const useUniV2LiquidityPool = ({
-    () => (skip ? null : getOrCreateContract(UNISWAP_V2_ROUTER_02_CONTRACT_ADDRESS)),
+    () => (skip ? null : getOrCreateContractByAddress(UNISWAP_V2_ROUTER_02_CONTRACT_ADDRESS)),
     [skip],
   )
 
   // Checksummed addresses
+  const asset0ContractAddress = useMemo(
+    () => ethers.utils.getAddress(fromAssetId(assetId0OrWeth).assetReference),
+    [assetId0OrWeth],
+  )
   const asset1ContractAddress = useMemo(
-    () => ethers.utils.getAddress(fromAssetId(assetId1).assetReference),
-    [assetId1],
+    () => ethers.utils.getAddress(fromAssetId(assetId1OrWeth).assetReference),
+    [assetId1OrWeth],
   )
 
   const lpContractAddress = useMemo(
     () => ethers.utils.getAddress(fromAssetId(lpAssetId).assetReference),
     [lpAssetId],
   )
 
+  const asset0Contract = useMemo(
+    () =>
+      skip
+        ? null
+        : getOrCreateContractByType({
+            address: asset0ContractAddress,
+            type: ContractType.ERC20,
+          }),
+    [asset0ContractAddress, skip],
+  )
+
   const asset1Contract = useMemo(
-    // TODO(gomes): remove casting and make getOrCreateContract handle generic ERC-20s as input
     () =>
-      skip ? null : getOrCreateContract(asset1ContractAddress as typeof FOX_TOKEN_CONTRACT_ADDRESS),
+      skip
+        ? null
+        : getOrCreateContractByType({
+            address: asset1ContractAddress,
+            type: ContractType.ERC20,
+          }),
     [asset1ContractAddress, skip],
   )
 
   const uniV2LPContract = useMemo(
-    // TODO(gomes): remove casting and make getOrCreateContract handle generic ERC-20s as input
     () =>
-      skip ? null : getOrCreateContract(lpContractAddress as typeof ETH_FOX_POOL_CONTRACT_ADDRESS),
+      skip
+        ? null
+        : getOrCreateContractByType({
+            address: lpContractAddress,
+            type: ContractType.UniV2Pair,
+          }),
     [lpContractAddress, skip],
   )
 
+  const makeAddLiquidityData = useCallback(
+    ({ token0Amount, token1Amount }: { token0Amount: string; token1Amount: string }) => {
+      if (!uniswapRouterContract) throw new Error('Uniswap router contract instance is undefined')
+      const deadline = Date.now() + 1000 * 60 * 10 // 10 minutes from now
+      if ([assetId0OrWeth, assetId1OrWeth].includes(wethAssetId)) {
+        const ethAmount = (() => {
+          if (assetId0OrWeth === wethAssetId) return token0Amount
+          if (assetId1OrWeth === wethAssetId) return token1Amount
+          return '0'
+        })()
+
+        const otherAssetContractAddress =
+          assetId0OrWeth === wethAssetId ? asset1ContractAddress : asset0ContractAddress
+        const otherAsset = assetId0OrWeth === wethAssetId ? asset1 : asset0
+        const otherAssetAmount = assetId0OrWeth === wethAssetId ? token1Amount : token0Amount
+
+        const accountAddress = fromAccountId(accountId).account
+
+        const amountOtherAssetMin = calculateSlippageMargin(otherAssetAmount, otherAsset.precision)
```

Comment:
this `calculateSlippageMargin` function has a hard coded 3% slippage 😬 - we can't have that high as a default

Reply:
Set to the default 0.5% UNI-V2 settings in 94a54a6896955feef3c15bcf667881cb21d27fba